src/main/java/io/kestra/plugin/azure/eventhubs/Consume.java

```diff
@@ -0,0 +1,249 @@
+@EqualsAndHashCode
+public class Consume extends AbstractEventHubTask implements EventHubConsumerInterface, RunnableTask<Consume.Output> {
+    // TASK'S PARAMETERS
+    @Builder.Default
+    private Serdes bodyDeserializer = Serdes.STRING;
+
+    @Builder.Default
+    private Map<String, Object> bodyDeserializerProperties = Collections.emptyMap();
+
+    @Builder.Default
+    private String consumerGroup = "$Default";
+
+    @Builder.Default
+    private StartingPosition partitionStartingPosition = StartingPosition.EARLIEST;
+
+    private String EnqueueTime;
+
+    @Builder.Default
+    private Integer maxBatchSizePerPartition = 50;
+
+    @Builder.Default
+    private Duration maxWaitTimePerPartition = Duration.ofSeconds(5);
+
+    @Builder.Default
+    private Duration maxDuration = Duration.ofSeconds(10);
+
+    @Builder.Default
+    private Map<String, String> checkpointStoreProperties = Collections.emptyMap();
+
+    // SERVICES
+    @Getter(AccessLevel.NONE)
+    @Builder.Default
+    private final EventHubClientFactory clientFactory = new EventHubClientFactory();
+
+    /**
+     * {@inheritDoc}
+     **/
+    @Override
+    public Output run(RunContext runContext) throws Exception {
+        return run(runContext, this);
+    }
+
+    /**
+     * Runs the consumer task using the specified context and plugin interface.
+     *
+     * @param runContext The context.
+     * @param task       The plugin interface.
+     * @return The output.
+     * @throws Exception if something wrong happens.
+     */
+    Output run(RunContext runContext, EventHubConsumerInterface task) throws Exception {
+
+        final DynamicEventHubConsumerConfig config = new DynamicEventHubConsumerConfig(runContext, task);
+
+        // Create converter
+        Serdes serdes = task.getBodyDeserializer();
+        Serde serde = serdes.create(task.getBodyDeserializerProperties());
+        EventDataObjectConverter converter = new EventDataObjectConverter(serde);
+
+        final EventHubConsumerService service = new EventHubConsumerService(
+            clientFactory,
+            config,
+            converter,
+            () -> getBlobCheckpointStore(runContext, task, clientFactory)
+        );
+
+        File tempFile = runContext.tempFile(".ion").toFile();
+        try (
+            BufferedOutputStream output = new BufferedOutputStream(new FileOutputStream(tempFile))
+        ) {
+
+            final AtomicReference<URI> uri = new AtomicReference<>();
+
+            Logger contextLogger = runContext.logger();
+
+            final ConsumerContext consumerContext = new ConsumerContext(
+                task.getMaxBatchSizePerPartition(),
+                task.getMaxWaitTimePerPartition(),
+                task.getMaxDuration(),
+                contextLogger
+            );
+
+            Map<EventHubNamePartition, Integer> result = service.poll(
+                consumerContext,
+                new EventHubConsumerService.EventProcessorListener() {
```

Comment:
This seems overly complicated to use a listener.
Can't we do without it? After all, the service can return the URI and that it.

Reply:
Yes, it's perfectly possible to do things differently, but I'm not sure it will be any simpler than it already is. The listener allows to only have to look to the task to understand how events are captured and stored without the need to digging into the consumer polling loop.